sql/src/main/java/io/crate/operation/projectors/SortingTopNProjector.java

```diff
@@ -96,15 +98,17 @@ public void startProjection() {
 
     @Override
     public synchronized boolean setNextRow(Row row) {
-        if (spare == null) {
```

Comment:
can you revert the changes in the SortingTopNProjector and instead create a separate OrderByPosition for the BucketMerger which is Row based and leave the Object[][] stuff here as is?